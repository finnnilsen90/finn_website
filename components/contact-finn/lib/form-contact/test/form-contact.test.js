const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Form = require('../form-contact.js');
const Button_one = require('../../../../lib/button-one/button-one.js');

let newForm = new Form();

describe('component connection', () => {

    it('should be defined', () => {
        expect(Form).toBeDefined();
    })

    it('button should be defined', () => {
        expect(Button_one).toBeDefined();
    })

})  
    
describe('<Form />', () => {

    const wrapper = render(<Form />);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe('Email response', () => {

    const wrapper = mount(<Form />, { attachTo: document.body });
    
    it('No message sent status successful', () => {
        expect(wrapper.state().msg_sent).toBe(false);
    })
    
    it('Message sent status successful', () => {
        wrapper.instance().message();
        expect(wrapper.state().msg_sent).toBe(true);
    })

    it('Error message handeling should exist', () => {
        let message = wrapper.find('.msg_comp');
        expect(message).toMatchSnapshot();
    })
    

    it('Should throw error if failure', () => {
        let wrapper_sent = shallow(<Form />);
        wrapper_sent.instance().message(false);
        let message = wrapper_sent.find('.msg_comp');
        expect(message).toMatchSnapshot();
    })
    
})