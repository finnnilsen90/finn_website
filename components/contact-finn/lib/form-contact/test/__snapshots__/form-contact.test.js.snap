// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders components 1`] = `
<div
  class="form-contact_Container font"
>
  <h1
    class="header"
  >
    Contact
  </h1>
  <form
    action="mailto:fnilsen1051@gmail.com"
    class="form"
    method="post"
  >
    <div
      class="input_box"
    >
      <label
        class="label e_label"
      >
        Email
      </label>
      <input
        class="input"
        name="email"
        type="text"
      />
      <br />
    </div>
    <div
      class="input_box"
    >
      <label
        class="label s_label"
      >
        Subject
      </label>
      <input
        class="input"
        name="subject"
        type="text"
      />
      <br />
    </div>
    <div
      class="input_box"
    >
      <label
        class="label message"
      >
        Message
      </label>
      <br />
      <textarea
        class="text_input"
        name="message"
        type="text"
      />
    </div>
    <div
      class="button"
    >
      <div
        class="button-one_Container"
      >
        <button
          class="button_style_one submit_msg"
          type="submit"
          value="Submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
